sdk/src/main/scala/com/horizen/SidechainNodeViewHolder.scala

```diff
@@ -24,7 +26,8 @@ class SidechainNodeViewHolder(sidechainSettings: SidechainSettings,
                               timeProvider: NetworkTimeProvider,
                               applicationWallet: ApplicationWallet,
                               applicationState: ApplicationState,
-                              genesisBlock: SidechainBlock)
+                              genesisBlock: SidechainBlock,
+                              certifier: Option[ActorRef])
```

Comment:
`Certifier` actor should not be a part of `SidechainNodeViewHolder`.
It must be a separate Actor, that should react on listen to `NodeViewSynchronizer.SemanticallySuccessfulModifier(mod)` event. Similar to the logic of `SidechainBlockActor`.
So, reacting on this event we need to:
1. Verify that certificate is not present in SC: for given block check that its the last of block of the epoch (all the data is in its blockinfo) or the block is somewhere in the middle of the epoch N and there is no cert for epoch N-1. So we can create a certificate.
2. Verify that Submitter has right to create a certificate (to be done in future probably)
3. Get all the data for Certificate (review the way of getting end mcblock hash)
4. *** verify that there is no cert you want to create already in the MC mempool (to be discussed)
5. Push the cert and mark it as "pushed", to not to duplicate certs.